lib/manageiq/api/common/headers.rb

```diff
@@ -0,0 +1,25 @@
+module ManageIQ
+  module API
+    module Common
+      module Headers
+
+        def self.current=(val)
+          validate_headers(val)
+          Thread.current[:attr_current_headers] = val
+        end
+
+        def self.current
+          Thread.current[:attr_current_headers]
+        end
+
+        def self.validate_headers(val)
+          if val.is_a?(ActionDispatch::Http::Headers)
+            true
+          else
+            raise ArgumentError, 'Not an ActionDispatch::Http::Headers Class'
+          end
```

Comment:
Minor style, but it's preferable to use guard clauses for validation...that is, this format is preferable...

```ruby
def self.validate_headers(val)
  raise ArgumentError, 'Not an ActionDispatch::Http::Headers Class' unless val.is_a?(ActionDispatch::Http::Headers)
  true
end
```

Reply:
Oh, also, should this probably be a private class method?

Reply:
Ha! - I thought our rubocop was against `unless` - I wanted to make it (had it) that way in the first place.

This makes me happy.  😄